### PR TITLE
Feature/select topics

### PR DESCRIPTION
**Pull request by rpannekoek**
* Introduced ability to select the topics to be shown on the home page
* Introduced ability to view statistics (min/max/avg) on the selected topics
* Refactored HTML generation (in particular: left menu HTML)
* Additional debug output for settings JSON

NOTE: This PR/branch is based on `feature/hotspotmode` and includes the code changes made for that branch. See this PR: https://github.com/Egyras/HeishaMon/pull/244

### CODE REVIEW

**Comment by IgorYbema** [comment]
@Egyras i'll take this into my repo for testing

**Comment by Egyras** [comment]
@IgorYbema ok, waiting outcome before clicking on merge ;)

**Comment by CurlyMoo** [comment]
Can you make this PR just about what you describe it's about. So, leaving all hotspot related changes out. I don't believe this PR is dependent on better hotspot functionality.

**Comment by rpannekoek** [comment]
It's indeed functionally not dependent, but building further on refactorings from the other PR.
I am using a combination of all changes on my own device.
Note that this PR is basically a combination of hotspot mode (#244) and topic selection; when this one is merged, #244 is automatically included.

**Comment by CurlyMoo** [comment]
> It's indeed functionally not dependent

Then i would advice to remove all non-dependent code from this PR, making this solely about the suggested changes. That increases the success of your changes being merged.

**Comment by IgorYbema** [comment]
I agree with @CurlyMoo 
As the PR's are now published makes it very hard to understand the dependencies, hard to understand what is going on, why modifications to not related code is made. And hard to merge with other alpha versions in my repo.

**Comment by CurlyMoo** [comment]
In general i would love some rationale for this PR. The HeishaMon is built to especially interact with domotica software. Not for standalone usage.

When thinking about changes to the GUI i would for example opt for letting you control the heatpump by directly changing the values on the home page. So you are able to quickly change something when necessary. I see the HeishaMon more as a _device driver_ instead of a full fledge standalone solution.

And in general, with all code changes and refactoring. Keep in mind memory usage and fragmentation. I would say that every PR should at least make memory usage and fragmentation improvements mandatory.

**Comment by rpannekoek** [comment]
Well, as described in the associated issue ( https://github.com/Egyras/HeishaMon/issues/246 ), I am using HeishaMon stand-alone myself and am quite happy with it.
I was just missing these features for that use case.
If this is not in line with your philosophy, and you don't want to incorporate this into the HeishaMon release, that's fine with me too.  You can just close the PR in that case.

**Comment by CurlyMoo** [comment]
> I am using HeishaMon stand-alone myself and am quite happy with it.

What does standalone in your case mean? So what does the HeishaMon offer, what other solutions couldn't? I'm honestly interested.

I also named some memory requirements that i think should be mandatory. Did you validate for those?

**Comment by trsqr** [comment]
> So what does the HeishaMon offer, what other solutions couldn't?

In my case it used to be like this (before I bothered enough to setup HomeAssistant):
I have a vacation home with Aquarea. I do rent it out sometimes. When it's not being rented out I lower the temperature. With HeishaMon I'm able to set the temperature to a lower setting remotely and increase it when needed. I also can read the current state of the pump, which gives me peace of mind. Also in case if the guests complain about low shower water temperature or such, I could see the current situation. What other solutions are there (unless you want to pay big money for Panasonic's shitty cloud-based system)? HomeAssistant is nice, but actually it didn't bring anything 100% necessary for this use case.

**Comment by rpannekoek** [comment]
HeishaMon allows me to monitor all aspects of my heatpump remotely. It provides me much more insights than the Remote Control panel.  Probably not more than Panasonic's Service Cloud, but I didn't buy the CZ-TAW because I saw that HeishaMon existed and it seemed a much more interesting alternative.

In order to obtain graphs (at daytime), I'm using a simple PowerShell script which periodically queries the data in JSON format and stores it in a CSV file.  The in-built statistics are in particular to monitor at nighttime (although they are nice in general).

BTW: Surely, I'm not the only one who wants to use HeishaMon in stand-alone mode. See https://github.com/Egyras/HeishaMon/issues/198

Regarding memory requirements: what are you worried about, in particular?  I'm running the code changes on my device and it has 66% memory free.  In general, I ensured to use PROGMEM in most places to keep RAM usage to a minimum. Keeping track of statistics takes 64 x 12 = 768 bytes of RAM.  For the refactored HTML generation, i'm using a 256 bytes/chars temporary buffer.

**Comment by IgorYbema** [comment]
Please stop these discussions in the PR. Go ahead to the slack channel for further discussions.

The point to make here is that adding too much changes to the code here is hard to follow. One feature per PR is better.  I do like the idea's presented but I can't maintain it this way.

**Comment by Egyras** [comment]
I can see nice discussions here, for standalone approach I suggest to make some lightweight phone app or PC app like this one which I started with my python learning https://github.com/Egyras/Panasonic_GUI. ESP8266 better to leave as pure controller without big changes in the current code. And Slack should be used instead PR comments :)
